File: 프로그래머스/1/42889. 실패율/실패율.py
from collections import defaultdict

def solution(N, stages):
    result = []
    length = len(stages)

    for i in range(1, N + 1):
        cnt = stages.count(i)
        result.append((i, cnt) if length <= 0 else (i, cnt / length))
        length -= cnt
    
    answer = sorted(result, key=lambda x: x[1], reverse=True)
    return [x[0] for x in answer]    
    

File: 프로그래머스/1/42889. 실패율/실패율.java
import java.util.*;

class Solution {
    
    public int[] solution(int N, int[] stages) {
        List<double[]> result = new ArrayList<>();
        int length = stages.length;

        for (int i = 1; i <= N; i++) {
            int cnt = 0;
            for (int stage : stages) {
                if (stage == i) cnt++;
            }

            if (length > 0) {
                result.add(new double[]{i, (double) cnt / length});
            } else {
                result.add(new double[]{i, 0});
            }

            length -= cnt;
        }

        result.sort((a, b) -> Double.compare(b[1], a[1]));

        int[] answer = new int[N];
        for (int i = 0; i < N; i++) {
            answer[i] = (int) result.get(i)[0];
        }

        return answer;
    }
}